engine/installation/linux/ubuntulinux.md

```diff
@@ -116,365 +119,501 @@ packages from the Docker repository:
+1.  Log into Ubuntu as a user with `sudo` privileges.
 
-2. Edit the `/etc/default/grub` file.
+2.  Edit the `/etc/default/grub` file.
 
-3. Set the `GRUB_CMDLINE_LINUX` value as follows:
+3.  Add or edit the `GRUB_CMDLINE_LINUX` line to add the following two key-value
+    pairs:
 
-        GRUB_CMDLINE_LINUX="cgroup_enable=memory swapaccount=1"
+    ```none
+    GRUB_CMDLINE_LINUX="cgroup_enable=memory swapaccount=1"
+    ```
 
-4. Save and close the file.
+    Save and close the file.
 
-5. Update GRUB.
+4.  Update GRUB.
 
-        $ sudo update-grub
+    ```bash
+    $ sudo update-grub
+    ```
 
-6. Reboot your system.
+     If your GRUB configuration file has incorrect syntax, an error will occur.
+     In this case, steps 3 and 4.
+
+6.  Reboot your system. Memory and swap accounting are enabled and the warning
+    does not occur.
 
 
 ### Enable UFW forwarding
 
 If you use [UFW (Uncomplicated Firewall)](https://help.ubuntu.com/community/UFW)
 on the same host as you run Docker, you'll need to do additional configuration.
 Docker uses a bridge to manage container networking. By default, UFW drops all
-forwarding traffic. As a result, for Docker to run when UFW is
-enabled, you must set UFW's forwarding policy appropriately.
+forwarding traffic. You must set UFW's forwarding policy appropriately.
 
-Also, UFW's default set of rules denies all incoming traffic. If you want to
-reach your containers from another host allow incoming connections on the Docker
-port. The Docker port defaults to `2376` if TLS is enabled or `2375` when it is
-not. If TLS is not enabled, communication is unencrypted. By default, Docker
-runs without TLS enabled.
+In addition, UFW blocks all incoming traffic by default. If you want to access
+the Docker Remote API from another host and you have enabled remote access, you
+need to configure UFW to allow incoming connections on the Docker port, which
+defaults to `2376` if TLS encrypted transport is enabled or `2375` otherwise. By
+default, Docker runs **without** TLS enabled. If you do not use TLS, you are
+strongly discouraged from allowing access to the Docker Remote API from remote
+hosts, to prevent remote privilege-escalation attacks.
 
 To configure UFW and allow incoming connections on the Docker port:
 
-1. Log into Ubuntu as a user with `sudo` privileges.
+1.  Log into Ubuntu as a user with `sudo` privileges.
+
+2.  Verify that UFW is enabled.
 
-2. Verify that UFW is installed and enabled.
+    ```bash
+    $ sudo ufw status
+    ```
 
-        $ sudo ufw status
+    If `ufw` is not enabled, the remaining steps will not be helpful.
 
-3. Open the `/etc/default/ufw` file for editing.
+3.  If you need to enable access to the Docker Remote API from external hosts
+    and understand the security implications (see the section before this
+    procedure) edit the UFW configuration file, which is usually
+    `/etc/default/ufw` or `/etc/sysconfig/ufw`. Set the `DEFAULT_FORWARD_POLICY`
+    policy to `ACCEPT`.
```

Comment:
Can you do a quick check with the network team if this only applies to the API? I think the warning may need to be on step 4., not here